File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int n = grid.size();
        vector<vector<int>> ans(n-2, vector<int>(n-2, 0));

        for(int i = 0; i<n-2; i++) {
            for(int j = 0; j<n-2; j++) {
                int val = INT_MIN;
                for(int x = i; x<i + 3; x++) {
                    for(int y = j; y<j + 3; y++) {
                        val = max(val, grid[x][y]);
                    }
                }
                ans[i][j] = val;
            }
        }

        return ans;
    }
};

File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.py
class Solution(object):
    def largestLocal(self, grid):
        n = len(grid)
        ans = [[0 for _ in range(n-2)] for _ in range(n-2)]

        for i in range(n-2):
            for j in range(n-2):
                val = float('-inf')
                for x in range(i, i + 3):
                    for y in range(j, j + 3):
                        val = max(val, grid[x][y])
                ans[i][j] = val
            
        return ans
        